src/main/java/com/alkemy/ong/service/impl/UserServiceImpl.java

```diff
@@ -104,4 +102,15 @@ else if(!passwordEncoder.matches(userReq.getPassword(),user.getPassword())){
         return userRepo.findAll();
     }
 
+    @Override
+    public UserRegisterResponse upgradeUser(Long id, UserRegisterRequest userJpa) throws NotFoundException {
+        if(!userRepo.existsById(id)){
+            throw new NotFoundException("The user is not registered.");
+        }
+            Users userBd = userRepo.findById(id).get();
```

Comment:
en este caso que necesitas quedarte con todo el objeto user, esta bueno que te lo traigas y depsues preguntas el IfPresent, en el caso del delete te hice sacarlo porque solo necesitabas saber si existia y nada mas.

De todas formas en este caso lo mejor que podes hacer para ahorrate el if y si no esta largar una exception es algo asi:
`Users userbd = userRepo.findById(id).orElseThrow( ()-> new NotFoundException("The user is not registered.")  );`

Haciendo esto te ahorras el if de mas arriba